src/server/middlewares/redirects.ts

```diff
@@ -25,21 +25,21 @@ export const forceHost = ({
   await next()
 }
 
-export const startPageRedirect: IMiddleware<object> = async (ctx) => {
-  const actualIp = ctx.ip
-  const fallbackCountryLabel = fallbackLocale.label
+export const geoRedirect = (
+  path: string,
+  fallbackCountryLabel = fallbackLocale.label,
+): IMiddleware<void> => async (ctx) => {
   const countryLabel =
-    lookupCountry(actualIp)?.toLowerCase() || fallbackCountryLabel
+    lookupCountry(ctx.ip)?.toLowerCase() || fallbackCountryLabel
   const queryStringMaybe = ctx.querystring ? '?' + ctx.querystring : ''
 
   ctx.status = 301
 
   if (Object.keys(locales).includes(countryLabel)) {
-    ctx.redirect(`/${countryLabel}${queryStringMaybe}`)
-    return
+    ctx.redirect(`/${countryLabel}${path}${queryStringMaybe}`)
+  } else {
+    ctx.redirect(`/${fallbackCountryLabel}${path}${queryStringMaybe}`)
```

Comment:
I kinda like not having an `else` but no biggie :)

Reply:
I used to but have changed my mind, imo `else`s are more verbose and obvious than early returns